pkg/client/event.go

```diff
@@ -0,0 +1,32 @@
+package client
+
+import (
+	"context"
+
+	apiv1 "github.com/acorn-io/acorn/pkg/apis/api.acorn.io/v1"
+)
+
+func (c *DefaultClient) EventStream(ctx context.Context, opts *EventStreamOptions) (<-chan apiv1.Event, error) {
+	listOpts := opts.ListOptions()
+	listOpts.Namespace = c.Namespace
+
+	var current apiv1.EventList
+	if err := c.Client.List(ctx, &current, listOpts); err != nil {
```

Comment:
yup, so I actually implemented that _first_

```go
func (c *DefaultClient) EventStream(ctx context.Context, opts *EventStreamOptions) (<-chan apiv1.Event, error) {
	listOpts := opts.ListOptions()
	listOpts.Namespace = c.Namespace

	var (
		current apiv1.EventList
		w       watch.Interface
		err     error
	)
	if opts.Watch {
		w, err = c.Client.Watch(ctx, &apiv1.EventList{}, &kclient.ListOptions{})
	} else {
		err = c.Client.List(ctx, &current, listOpts)
	}
	if err != nil {
		return nil, err
	}

	events := make(chan apiv1.Event)
	go func() {
		defer close(events)
		// Send the current set of events
		for _, c := range current.Items {
			select {
			case <-ctx.Done():
				return
			case events <- c:
			}
		}

		if w == nil {
			// Don't need to watch
			return
		}

		defer w.Stop()

		for {
			select {
			case <-ctx.Done():
				return
			case e := <-w.ResultChan():
				switch e.Type {
				case kwatch.Error:
					return
				case kwatch.Added, kwatch.Modified:
					events <- *(e.Object.(*apiv1.Event))
				}
			}
		}
	}()

	return events, nil
}
```

I decided to roll this back because I figured it'd be contentious + table formatting gets a bit funky when we start flushing multiple times.

As it stands, I'm _very_ on the fence; should I just implement the vanilla list client instead?

Reply:
I think it's fine.